src/components/edit-movie-form/edit-movie-form.tsx

```diff
@@ -1,105 +1,205 @@
+import { API_BASE } from '@constants';
 import { Button } from 'components/button';
 import { Input } from 'components/input';
 import { ReleaseDatePicker } from 'components/release-date-picker';
 import { Select } from 'components/select';
-import React, { FC, FormEvent, useState } from 'react';
+import { useFormik } from 'formik';
+import { useApiRequest } from 'hooks/useApiRequest';
+import moment from 'moment';
+import React, { FC } from 'react';
+import { useSelector } from 'react-redux';
+import { editMovie } from 'redux/actions';
+import * as Yup from 'yup';
 import {
   CloseButton,
   EditMovieButtonContainer,
   EditMovieFormContainer,
+  EditMovieFormError,
   EditMovieFormInner,
   EditMovieFormTitle,
   EditMovieFormWrapper,
 } from './styles';
-
 interface EditMovieFormProps {
   hideEdit: () => void;
 }
 
-const initialValues = {
-  genre: '',
-  movie: '',
+const initialValue = {
+  id: '',
+  title: '',
+  release_date: '',
+  poster_path: '',
+  genres: [],
   overview: '',
-  rating: '',
   runtime: '',
-  title: '',
 };
 
-export const EditMovieForm: FC<EditMovieFormProps> = ({ hideEdit }) => {
-  const [values, setValues] = useState(initialValues);
+const validationSchema = Yup.object({
+  id: Yup.string().required('Required'),
+  title: Yup.string().required('Required'),
+  poster_path: Yup.string()
+    .required('Required')
+    .url('The "Movie url" field is not a valid URL.'),
+  overview: Yup.string().required('Required'),
+  runtime: Yup.number()
+    .required('Required')
+    .typeError('The "Runtime" field must be a Number.')
+    .positive('The "Runtime" field must be a Positive Number.'),
+  genres: Yup.array().min(1, 'The "Genres" field must have at least 1 items'),
+});
 
-  const handleOnChange = ({ target }) => {
-    const value = target.type === 'checkbox' ? target.checked : target.value;
+export const EditMovieForm: FC<EditMovieFormProps> = ({ hideEdit }) => {
+  const posterId = useSelector(({ movies: { posterId } }) => posterId);
+  const movie = useSelector(({ movies: { items } }) =>
+    items.find((movie) => movie.id === posterId),
+  );
+  const initialValues = { ...initialValue, ...movie };
+  const { fetchData: fetchEditMovie } = useApiRequest(
+    'put',
+    API_BASE,
+    editMovie,
+  );
 
-    setValues({
+  const onSubmit = (values) => {
+    const body = {
       ...values,
-      [target.name]: value,
-    });
+      runtime: parseInt(values.runtime),
+      id: parseInt(values.id),
```

Comment:
for parseInt, it is better always to write a second radix parameter
https://newbedev.com/why-do-we-need-to-use-radix-parameter-when-calling-parseint